llama-index-integrations/vector_stores/llama-index-vector-stores-tablestore/llama_index/vector_stores/tablestore/base.py

```diff
@@ -0,0 +1,504 @@
+            return tablestore.MatchAllQuery()
+        return self._parse_filters_recursively(filters=filters)
+
+    @staticmethod
+    def _parse_filter(filter_item: MetadataFilter) -> tablestore.Query:
+        key = filter_item.key
+        val = filter_item.value
+        op = filter_item.operator
+
+        if op == FilterOperator.EQ:
+            return tablestore.TermQuery(field_name=key, column_value=val)
+        elif op == FilterOperator.GT:
+            return tablestore.RangeQuery(
+                field_name=key, range_from=val, include_lower=False
+            )
+        elif op == FilterOperator.GTE:
+            return tablestore.RangeQuery(
+                field_name=key, range_from=val, include_lower=True
+            )
+        elif op == FilterOperator.LT:
+            return tablestore.RangeQuery(
+                field_name=key, range_to=val, include_upper=False
+            )
+        elif op == FilterOperator.LTE:
+            return tablestore.RangeQuery(
+                field_name=key, range_to=val, include_upper=True
+            )
+        elif op == FilterOperator.NE:
+            bq = tablestore.BoolQuery()
+            bq.must_not_queries.append(
+                tablestore.TermQuery(field_name=key, column_value=val)
+            )
+            return bq
+        elif op in [FilterOperator.IN, FilterOperator.ANY]:
+            return tablestore.TermsQuery(field_name=key, column_values=val)
+        elif op == FilterOperator.NIN:
+            bq = tablestore.BoolQuery()
+            bq.must_not_queries.append(
+                tablestore.TermsQuery(field_name=key, column_values=val)
+            )
+            return bq
+        elif op == FilterOperator.ALL:
+            bq = tablestore.BoolQuery()
+            for val_item in val:
+                bq.must_queries.append(
+                    tablestore.TermQuery(field_name=key, column_value=val_item)
+                )
+            return bq
+        elif op == FilterOperator.TEXT_MATCH:
+            return tablestore.MatchQuery(field_name=key, text=val)
+        elif op == FilterOperator.CONTAINS:
+            return tablestore.WildcardQuery(field_name=key, value=f"*{val}*")
+        else:
+            raise ValueError(f"Unsupported filter operator: {filter_item.operator}")
+
+    @property
+    def client(self) -> Any:
+        """Get client."""
+        return self._tablestore_client
+
+    def add(self, nodes: List[BaseNode], **kwargs: Any) -> List[str]:
+        """Add nodes to vector store."""
+        if len(nodes) == 0:
+            return []
+        ids = []
+        for node in nodes:
+            self._write_row(
+                row_id=node.node_id,
+                content=node.text,
+                embedding_vector=node.get_embedding(),
+                metadata=node.metadata,
+            )
+            ids.append(node.node_id)
+        return ids
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
```

Comment:
This is meant to delete any row that has the given `ref_doc_id` in the metadata. Is this possible?

Deleting (and getting) by ID should be implemented with `get_nodes()` and `delete_nodes()` from the base class